Bottom: 4809cbe88afb1813e1fec6dec440d97999a07f6f
Top:    f5f193004d0e86eeb90acf61145b7fe193f1a9eb
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:24:09 +0800

autofs-5.0.7 - add null check in parse_server_string()

Add an error check for the case there's no ':' server name delimiter in
parse_server_string().


---

diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 17cbe9a..a59de92 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -1212,6 +1212,11 @@ static int parse_server_string(unsigned logopt, const char *url, struct lookup_c
 		/* Isolate the server. Include the port spec */
 		if (*ptr != '[')
 			q = strchr(ptr, ':');
+			if (!q) {
+				crit(logopt, MODPREFIX
+				     "LDAP server name not found in %s", ptr);
+				return 0;
+			}
 		else {
 			q = ++ptr;
 			while (*q == ':' || isxdigit(*q))
